internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -2887,11 +2910,91 @@ func validateCABundle(bundle string) error {
 	return nil
 }
 
+// validateImmutableAnnotations ensures that prefix annotations are immutable once set
+// Returns true if the immutable annotations are valid, false if they are invalid, and an error if there is an error
+func validateImmutableAnnotations(ctx context.Context, log logrus.FieldLogger, asc ASC) (bool, error) {
+	if _, ok := asc.Object.(*aiv1beta1.AgentServiceConfig); !ok {
+		// if status is nil, is not AgentServiceConfig object
```

Comment:
Wait sorry I forgot to comment on this as I'm not sure the comment matches the check? 😅

Reply:
Oh sorry, I've changed to type check as it was more explicit but didn't change the comment

Reply:
Ah gotcha, no need to be sorry!